### PR TITLE
Display source directory instead of need directory

### PR DESCRIPTION
**Pull request by vmrob**

### CODE REVIEW

**Comment by ccbrown** [comment]
Conceptually, I think the *library* is in dev mode. Not the library's source. So it makes sense to me if the URI displayed for the status is for the library, not the source. It might be the case that for your workflow, you usually copy and paste the output of this command with the intention of navigating to the source directory. But I don't think this is the place to make assumptions about that. It is more accurate to show the URI representative of the thing that's actually in dev mode.

**Comment by vmrob** [comment]
I think that's a fair point, but it's also worth noting that the contents of the directories inside `need_directory` is supposed to be entirely opaque. At no time should a developer really need to be aware of what exists inside the entire needs hierarchy except when a library is in dev mode. And then, the developer doesn't need to be aware of the build process (except for debugging) nor the downloads directory.

TLDR, I think the only directory that should be relevant to the developer is the source directory.

**Comment by ccbrown** [comment]
I don't think the TLDR is true, and it's the kind of thing that we certainly can't count on holding true as new features are added. There will be more things of interest as features are added, and I'd prefer more scalable ways of locating them.

I agree that ideally the need's directory structure would be opaque, but I think it's okay for the URI for the need to be opaque too.

So. Alternatives I think I would be more in favor of...

* A flag that changes the behavior:

  `-s, --source     shows source directories and source status messages`

  ```
% needy dev status -s
There is 1 library in dev mode:

  live  /Users/cbrown/Development/live-swarm-server/needs/live/source  4 modified files
```

  More informative, and scales well with features. I think there are other details the default and other potential flags might want to show.

* A subcommand that shows the source directory for a need:

  ```
% needy dev sourcedir live
/Users/cbrown/Development/live-swarm-server/needs/live/source
```

  Handy for plumbing.

**Comment by vmrob** [comment]
I can't think of a personal use case where I would use `needy dev status` and not want the `-s` flag if only for the additional information provided. I'm basically thinking the source status, in dev mode, is probably more relevant than the need status. It makes me think a `needy status` would be more desirable to provide that kind of need-level information.

```
$ needy status mylib
[UP-TO-DATE] mylib  /Users/me/dev/org/repo/needs/mylib
    source: git@github.com:org/repo
    commit: 1234567890ABCDEF
```

```
$ needy dev status mylib
mylib is in dev mode: /Users/me/dev/org/repo/needs/mylib/source
    source: git@github.com:org/repo
    <source-specific information, `git status`?>
```

And then for extensibility...

```
$ needy exec mylib git status
On branch develop
Your branch is up-to-date with 'origin/develop'.
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   src/modified-file.cpp

no changes added to commit (use "git add" and/or "git commit -a")
```

**Comment by ccbrown** [comment]
It's not in the slightest bit about your current personal use-cases. This is a firm stance against representing a need with any of its subdirectories.

I'm not against any of your other ideas. Giving `dev status` more detailed, non-tabular output is fine, but if you include the source directory there, pretend you don't know all of our future features and use-cases, and allow for other types of needs to potentially show other information instead.
